Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
I think it's really important to see this in context https://github.com/thoughtbot/Argo/blob/master/Argo/Operators/Operators.swift#L16-L18